index.js

```diff
@@ -1,10 +1,115 @@
 const express = require('express')
 const app = express()
-const port = 3000
-var cowsay = require("cowsay");
-
-console.log(cowsay  .say({
-    text : "Hello World",
-    e : "^^",
-    T : "U "
-}));
+const port = process.env.port || 3000
+
+// File System
+const fs = require('fs');
+
+//Body Parser
+const bodyParser= require('body-parser')
+app.use(bodyParser.urlencoded({extended:false}))
+app.use(bodyParser.json())
+
+// Data layer
+const FILE_PATH = './DataLayer/podcasts.json'
+let podcastData = require(FILE_PATH)
+
+//Validator
+const joi = require('joi')
+
+app.get('/podcast/:id',(req,res)=>{
+    const queryParams = req.params
+    const {error} = ValidateInput(queryParams)
+    if(error) return res.status(400).send(error.details[0].message)
+try{
+    const podcastInfo = getPodcastById(queryParams.id)
+    return res.status(200).send(podcastInfo)
+    }
+catch(err){
+    console.log(`Error: ${err.stack}`)
+    res.status(500).send(`Server Error: ${err.message}`)
+}
+})
+
+app.post('/podcast/new',(req,res)=>{
+    if(getPodcastById(req.body.id).length === 0){
+         try{
+             savePodcast(req.body)
+         }
+         catch(err){
+             console.log(`Error: ${err.stack}`)
+             res.status(500).send(`Server file error: ${err.message}`)
+         }
+        return res.status(200).send("Podcast has been successfully added")
+    }
+    else{
+        return res.status(409).send("Unable to add, podcast already exists!")
+    }
+})
+
+app.delete('/podcast/:id',(req,res)=>{
+    const queryParams = req.params
+    const {error} = ValidateInput(queryParams)
+    if(error) return res.status(400).send(error.details[0].message)
+    try{
+        const podcastInfo = getPodcastById(queryParams.id)
+        if(podcastInfo.length !== 0){//none empty array
+            deletePodcast(queryParams.id)
+            return res.status(200).send('Podcast has been successfully removed')
+        }
+        else{
+            res.status(404).send('Podcast not found')
+        }
+    }
+    catch(err){
+        console.log(`Error: ${err.stack}`)
+        res.status(500).send(`Server Error: ${err.message}`)
+    }
+})
+
+app.put('/podcast/:id',(req,res)=> {
+    if(getPodcastById(req.params.id).length !== 0){
+        try{
+            updatePodcasts(req.body,req.params.id)
+        }
+        catch(err){
+            console.log(`Error: ${err.stack}`)
+            res.status(500).send(`Server file error: ${err.message}`)
```

Comment:
If you reach the catch block two responses will be sent - the one in line 77 and the one on line 79 (because there is no return before res.status in line 77). In general the best practice is to always use return before res.send